### PR TITLE
Feature/block user

### PR DESCRIPTION
**Pull request by sylvanshine**
Block user functionality is ready for review - 
#34  - Blocked users 
#52 - zero users shown when user initiates private chat

### CODE REVIEW

File: src/main/java/com/graj/chatProject/functionGroups/users/UserService.java [hunk 1]
```diff
@@ -190,17 +219,65 @@ public UserContext prepareUserContext(String userName){
...
+      case NOT_BLOCKED:
+        update.pull("blockedUsers",contactId);
+        break;
+    }
+
+    mongoOps.updateFirst(query,update,User.class);
+  }
+
+  /**
+   * Abstraction to allow for change in bcloed users data if needed
```
Comment (tomssski) [inline src/main/java/com/graj/chatProject/functionGroups/users/UserService.java]: typo: 'bcloed' instead of 'blocked'

File: src/styles.scss [hunk 2]
```diff
@@ -1,41 +1,5 @@
 /* You can add global styles to this file, and also import other style files */
-@import "~@angular/material/prebuilt-themes/indigo-pink.css";
-
 body{
   background-color: aliceblue;
 }
 
-// adds enter icon to an element, e.g. input or dropdown option
```
Comment (tomssski) [inline src/styles.scss]: below code is missing in the merged file, but is needed

File: src/app/components/chat-central/chat-userlist/chat-userlist.component.html [hunk 3]
```diff
@@ -1,7 +1,21 @@
-<h5>{{ userCount }} Members</h5>
+<h5>{{ users.length }} Members</h5>
 <ul class="list-unstyled userlist">
   <li *ngFor="let user of users">
-    <i class="fa fa-user text-primary"></i>
-    {{ user.userName }}
+    <button class="btn btn-block text-left" style="background-color: transparent" [matMenuTriggerFor]="appMenu">
```
[the remark this answers is not in the document]
Comment (sylvanshine) [reply src/app/components/chat-central/chat-userlist/chat-userlist.component.html]: Done

File: src/app/services/contacts/contacts.service.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  public removeContactGlobalData(userId: String) {
+    this.contacts.delete(userId);
+  }
+
+  /**
+   * Gets the list of users that the current user has blocked
+   */
+  public updateBlockedUsersFromServer() {
+    this._httpClient.get('/user/blocked-users').subscribe( response => {
+      // this.blockedUsers = response ;
```
[the remark this answers is not in the document]
Comment (sylvanshine) [reply src/app/services/contacts/contacts.service.ts]: Done

**Pull request merged**